pkg/proxy/server.go

```diff
@@ -72,12 +80,30 @@ func NewServer(ctx context.Context, c CompletedConfig) (*Server, error) {
 		},
 	)
 
-	s.Handler, err = NewHandler(ctx, s.CompletedConfig.Options, s.IndexController)
-	if err != nil {
-		return s, err
+	if c.Options.UserGating.Enabled {
```

Comment:
`if c.Options.UserGating.DefaultRulesFile != ""` to be able to get rid of the bool

Reply:
I didn't make the static file based rules required since the primary ruleset lives in some workspace where other systems can dynamically update it.